File: 2025_Oct/evelyn/week3/2025-11-03-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int ans = INT_MIN;
        int left = 0;
        int right = height.size() - 1;

        while (right >left) {
            ans = max(ans, (right-left) * min(height[left], height[right]));

            if (height[right] > height[left]) left++;
            else right--;
        }

        return ans;
    }
};

File: 2025_Oct/JusungPark/2025-11-03-Container-With-Most-Water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int max = 0;

        while (left < right) {
            int h = Math.min(height[left], height[right]);
            int w = right - left;
            max = Math.max(max, h * w);

            if (height[left] < height[right]) left++;
            else right--;
        }
        return max;
    }
}
